### PR TITLE
Move hourly maintenance logic into a checked-in script

### PR DESCRIPTION
**Pull request by fejta**
Let's move the complexity into a checked in file rather than a jenkins text box (even if the value of that text box is controlled by a yaml file)

### CODE REVIEW

File: jenkins/hourly_maintenance.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import argparse
+import collections
+import datetime
+import re
+import subprocess
+import sys
+
+
+def ContainerImages():
+  for line in subprocess.check_output([
```
Comment (rmmh) [inline jenkins/hourly_maintenance.py]: please use 4-space indents for consistency

File: jenkins/hourly_maintenance.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  now = datetime.datetime.now()
+  old = []
+  for line in subprocess.check_output([
+      'docker',
+      'ps', '-a',
+      '-f', 'status=created',  # Never started due to timeout
+      '-f', 'status=exited',  # Container exited
+      '-f', 'status=dead',  # Zombie container
+      '--format={{.CreatedAt}}\t{{.ID}}\t{{.Image}}',
+  ]).split('\n'):
```
Comment (rmmh) [inline jenkins/hourly_maintenance.py]: Can you guarantee that this is run in a UTC environment? I'm a little worried about the 2 hour limit interacting with a PDT timezone and experiencing skew. You can override the TZ in the subprocess invocation.

**Comment by rmmh** [comment]
LGTM % nits

File: jenkins/hourly_maintenance.py (hunk 2 above)
Comment (fejta) [inline jenkins/hourly_maintenance.py]: I'm assuming the timezone is the same for the python and docker process.

File: jenkins/hourly_maintenance.py (hunk 1 above)
Comment (fejta) [inline jenkins/hourly_maintenance.py]: Done

**Pull request merged**